### PR TITLE
Transport list

### PR DESCRIPTION
**Pull request by mo-c4t**
* Add transport list service
* Add schedule for the whole trip in Trip message

Diagram: https://storage.googleapis.com/docs-cmp-files/diagrams/draft/proto/cmp/services/transport/v1/list.proto.dot.xs.svg

### CODE REVIEW

**Review by SamJaarsma — APPROVED** [review]
LGTM

File: proto/cmp/services/transport/v3/search.proto [hunk 1]
```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+
+// ## Transport Service
+//
+// The Transport Service is used for flights, trains, and buses.
+//
+// Any search message response in the Camino Messenger Protocol only includes
+// dynamic data. Static data can be cached and kept up to data with the Product List
```
Comment (havan) [inline proto/cmp/services/transport/v3/search.proto]:
```suggestion
// dynamic data. Static data can be cached and kept up to date with the Product List
```
Typo

File: proto/cmp/services/transport/v3/trip_types.proto [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+  // Luggage
+  //
+  // Ex: `3`
+  cmp.types.v1.Baggage baggage = 2;
+
+  // Price
+  //
+  // Ex: `Price(net=240, currency=Currency.CURRENCY_EUR)
+  cmp.types.v2.Price price = 3;
```
Comment (havan) [inline proto/cmp/services/transport/v3/trip_types.proto:31]: Previously we had the `price` in the `TripSegment`. Which means it was possible to define prices for individual segments. And now it's not. Is that intentional?

File: proto/cmp/services/transport/v3/trip_types.proto [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  string carrier_code = 1;
+
+  // Carrier Type. Ex: airline, transfer company, train company, .. etc
+  string carrier_type = 2;
+
+  // Start and end datetime for the trip
+  cmp.types.v1.DateTimeRange period = 3;
+
+  // Schedule for each segment of the trip
+  repeated cmp.services.transport.v3.SegmentInfo segments = 4;
```
Comment (havan) [inline proto/cmp/services/transport/v3/trip_types.proto]: We have the segments in the `Trip` type and also here. Why is that? Isn't that means we have the same info in two different places in the same service?

File: proto/cmp/services/transport/v3/trip_types.proto [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+  // Service type description examples:
+  // flight: "Economy Class", "Business Class", "First Class"
+  // train: "First Class", "Second Class" or "standard", "comfort", and "premium",..
+  // transfer: "Shuttle", "Private", "VIP Limosine", "Speedy Shuttle",...
+  string service_type_description = 3;
+
+  // Included, optional or compulsary services for this segment or leg. Just a code and
+  // description can be included if it is a package (seat, priority, large carry-on).
+  // or a price detail can be included for optionally pre-ordered meals for example.
```
Comment (havan) [inline proto/cmp/services/transport/v3/trip_types.proto]:
```suggestion
  // Included, optional or compulsory services for this segment or leg. Just a code and
  // description can be included if it is a package (seat, priority, large carry-on).
  // or a price detail can be included for optionally pre-ordered meals for example.
```
Typo

File: proto/cmp/services/transport/v3/trip_types.proto [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  // Retailer Code, replacing Marketing carrier The example would be where Iberia is
+  // selling an Iberia Express flight. So in this example the retailer_code for
+  // Iberia would be "IB" and the provider_code for Iberia Express would be "I2".
+  // Similar "interline" or "code share" operation may exist in rail and transfer.
+  string retailer_code = 3;
+
+  // Sub supplier code, replacing Operating carrier or identifying sub contracted
+  // services provided by other suppliers.
+  //
+  // DB could be selling an intenational train trip operated by the Dutch Nederlands
```
Comment (havan) [inline proto/cmp/services/transport/v3/trip_types.proto]:
```suggestion
  // DB could be selling an international train trip operated by the Dutch Nederlands
```
Typo

**Review by havan — COMMENTED** [review]
I have some comments about the new Trip types and some minor typos.

File: proto/cmp/services/transport/v3/trip_types.proto (hunk 3 above)
Comment (havan) [reply proto/cmp/services/transport/v3/trip_types.proto]: Oh, this is `SegmentInfo` type, not `TripSegment`. This is confusing, why we have segment info in the trip schedule type?

File: proto/cmp/services/transport/v3/search.proto [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+  // Search response metadata
+  cmp.types.v2.SearchResponseMetadata metadata = 2;
+
+  // Conten source types for this search request.
+  //
+  // Ex: ContentSourceType.CONTENT_SOURCE_TYPE_GDS,
+  // ContentSourceType.CONTENT_SOURCE_TYPE_NDC
+  // ContentSourceType.CONTENT_SOURCE_TYPE_3RD_PARTY
+  repeated cmp.types.v1.ContentSourceType conten_source_types = 3;
```
Comment (havan) [inline proto/cmp/services/transport/v3/search.proto]:
```suggestion
  // Content source types for this search request.
  //
  // Ex: ContentSourceType.CONTENT_SOURCE_TYPE_GDS,
  // ContentSourceType.CONTENT_SOURCE_TYPE_NDC
  // ContentSourceType.CONTENT_SOURCE_TYPE_3RD_PARTY
  repeated cmp.types.v1.ContentSourceType content_source_types = 3;
```
Typo

File: proto/cmp/services/transport/v3/search_result_types.proto [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+import "cmp/types/v2/cancel_policy.proto";
+import "cmp/types/v2/change_policy.proto";
+import "cmp/types/v2/price.proto";
+
+// Transport search result
+//
+// ![Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v2/search_result_types.proto.dot.xs.svg)
+//
+// [Open Message Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v2/search_result_types.proto.dot.svg)
+message TransportSearchResult {
```
Comment (havan) [inline proto/cmp/services/transport/v3/search_result_types.proto]:
```suggestion
// Transport search result
//
// ![Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v3/search_result_types.proto.dot.xs.svg)
//
// [Open Message Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v3/search_result_types.proto.dot.svg)
message TransportSearchResult {
```
Wrong version in diagram URL.

File: proto/cmp/services/transport/v3/search.proto [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+// Transport Search Service definition.
+//
+// Takes `TransportSearchRequest` message type and returns `TransportSearchResponse`
+// message type.
+//
+// ![Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v2/search.proto.dot.xs.svg)
+//
+// [Open Message Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v2/search.proto.dot.svg)
+service TransportSearchService {
```
Comment (havan) [inline proto/cmp/services/transport/v3/search.proto]:
```suggestion
// Takes `TransportSearchRequest` message type and returns `TransportSearchResponse`
// message type.
//
// ![Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v3/search.proto.dot.xs.svg)
//
// [Open Message Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/services/transport/v3/search.proto.dot.svg)
service TransportSearchService {
```
Wrong version in diagram URLs.

**Review by havan — CHANGES_REQUESTED** [review]
There are a typo and wrong versions diagram links.

**Review by havan — APPROVED** [review]

**Pull request merged**